integration/singularity/store.go

```diff
@@ -290,31 +293,51 @@ func (s *SingularityStore) runPreparationJobs() {
 
 	for {
 		select {
+		// If context is cancelled, end
 		case <-s.closing:
 			return
+
+		// If a new file came in, prepare it for packing, and mark the source
+		// ready to pack if the threshold is reached
 		case fileID := <-s.toPack:
-			prepareToPackSourceRes, err := s.singularityClient.File.PrepareToPackFile(&file.PrepareToPackFileParams{
+			prepareToPackFileRes, err := s.singularityClient.File.PrepareToPackFile(&file.PrepareToPackFileParams{
 				Context: ctx,
 				ID:      int64(fileID),
 			})
 			if err != nil {
 				logger.Errorw("preparing to pack file", "fileID", fileID, "error", err)
 			}
-			if prepareToPackSourceRes.Payload > s.packThreshold {
-				// mark outstanding pack jobs as ready to go so we can make CAR files
-				_, err := s.singularityClient.Job.PrepareToPackSource(&job.PrepareToPackSourceParams{
-					Context: ctx,
-					ID:      s.preparationName,
-					Name:    s.sourceName,
-				})
-				if err != nil {
+			if prepareToPackFileRes.Payload > s.packThreshold {
+				if err := s.prepareToPackSource(ctx); err != nil {
 					logger.Errorw("preparing to pack source", "error", err)
 				}
 			}
+
+		// If forced pack message comes through (e.g. from pack threshold max
+		// wait time being exceeded), prepare to pack source immediately
+		case <-s.forcePack.C:
```

Comment:
We need to handle a few more cases here:
1. If new files keep coming and new pack jobs are being marked as ready, we should extend the forcePack timer
2. If new files keep coming but they are so small and won't fill up a packJob for days, should we do forcePack (I'm leaning toward yes)

Reply:
also, what would be a good default value?

i'd probably rename PACK_THRESHOLD_MAX_WAIT to FORCE_PACK_AFTER

```
FORCE_PACK_AFTER=24h
FORCE_PACK_BELOW_HOURLY_RATE=256MiB
```

?

Reply:
My opinion is to just implement FORCE_PACK_AFTER if there is no packJob being marked for last X hours. Hourly rate is a good idea but I would defer it until we hear some feedback from Beta users.
Curious about what others think. @masih @willscott

Reply:
ok, updated with that behavior for now.